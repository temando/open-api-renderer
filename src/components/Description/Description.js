import React, { PureComponent } from 'react'
import injectSheet, { withTheme } from 'react-jss'
import classNames from 'classnames'
import markdown from 'markdown-it'
import PropTypes from 'prop-types'
import { styles } from './Description.styles'

const cm = markdown('commonmark')

@withTheme
@injectSheet(styles)
export default class Description extends PureComponent {
  render () {
    const { isInline, description, classes } = this.props

    let text
    if (isInline) {
      text = {
        __html: cm.renderInline(description)
      }
    } else {
      text = {
        __html: cm.render(description)
      }
    }

    return (
      <div className={classNames(classes.description, {
        [classes.inline]: isInline
      })} dangerouslySetInnerHTML={text} />
    )
  }
}

Description.propTypes = {
  description: PropTypes.string.isRequired,
  isInline: PropTypes.bool,
  classes: PropTypes.object
}
