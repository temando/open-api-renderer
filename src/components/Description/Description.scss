.description-inline,
.description-inline p {
  display: inline;
}

.description p {
  margin: .5rem 0;
}
