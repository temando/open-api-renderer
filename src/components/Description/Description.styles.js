import c from 'color'

export const styles = theme => ({
  'description': {
    color: `${c(theme.text.description)}`,
    maxWidth: '80rem',

    '&$inline': {
      fontSize: `${theme.sizes.text}`,
      '&, & p': {
        display: 'inline'
      }
    },
    '& p': {
      margin: '.5rem 0'
    },
    '& p, & p *': {
      fontSize: `${theme.sizes.text}`
    }
  },
  'inline': {
    paddingRight: '.2rem'
  }
})
