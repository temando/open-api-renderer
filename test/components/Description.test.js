import React from 'react'
import Description from './../../src/components/Description/Description'
import renderer from 'react-test-renderer'

describe('<Description />', () => {
  it('renders text with no commonmark as html', () => {
    const text = 'This method has zero markdown.'
    const tree = renderer.create(
      <Description description={text} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders commonmark text as html', () => {
    const text = 'This method has some `var i = 0` _markdown_, including a [link](http://www.google.com).'
    const tree = renderer.create(
      <Description description={text} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders text inline if asked', () => {
    const text = 'This method has zero markdown.'
    const tree = renderer.create(
      <Description isInline description={text} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
