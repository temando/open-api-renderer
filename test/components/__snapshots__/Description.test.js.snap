// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> renders commonmark text as html 1`] = `
<div
  className=""
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>This method has some <code>var i = 0</code> <em>markdown</em>, including a <a href=\\"http://www.google.com\\">link</a>.</p>
",
    }
  }
/>
`;

exports[`<Description /> renders text inline if asked 1`] = `
<div
  className="undefined"
  dangerouslySetInnerHTML={
    Object {
      "__html": "This method has zero markdown.",
    }
  }
/>
`;

exports[`<Description /> renders text with no commonmark as html 1`] = `
<div
  className=""
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>This method has zero markdown.</p>
",
    }
  }
/>
`;
